libr/core/windows_heap.c

```diff
@@ -60,14 +62,42 @@
 	}
 
 #define UPDATE_FLAGS(hb, flags)\
-	if ((flags & 0xf1) || (flags & 0x0200)) {\
+	if (((flags) & 0xf1) || ((flags) & 0x0200)) {\
 		hb->dwFlags = LF32_FIXED;\
-	} else if ((flags & 0x20)) {\
+	} else if ((flags) & 0x20) {\
 		hb->dwFlags = LF32_MOVEABLE;\
-	} else if ((flags & 0x0100)) {\
+	} else if ((flags) & 0x0100) {\
 		hb->dwFlags = LF32_FREE;\
 	}\
-	hb->dwFlags |= (flags >> SHIFT) << SHIFT;\
+	hb->dwFlags |= ((flags) >> SHIFT) << SHIFT;
+
+#define FILL_TYPE(type, flags)\
+	memset (type, 0, sizeof (type));\
+	switch (flags & 0xFFFF) {\
+	case LF32_FIXED:\
+		strncpy (type, "(FIXED)", 8);\
+		break;\
+	case LF32_FREE:\
+		strncpy (type, "(FREE)", 7);\
+		break;\
+	case LF32_MOVEABLE:\
+		strncpy (type, "(MOVEABLE)", 11);\
+		break;\
+	}\
+	if (flags & SEGMENT_HEAP_BLOCK) {\
+		strncat (type, "Segment", 8);\
```

Comment:
Strncat is a pretty bad call and shoukdnt be used. Also its used incorrectly. Better use rstr apis which are safer and simpler, like strnewf and strappendf..